zaza/openstack/utilities/openstack.py

```diff
@@ -2501,6 +2540,50 @@ def create_image(glance, image_url, image_name, image_cache_dir=None, tags=[],
     return image
 
 
+def create_import_image(glance, image_url, image_name, image_cache_dir=None,
+                        backend=None, disk_format='qcow2', visibility='public',
+                        container_format='bare'):
+    """Download the image and upload it to glance.
+
+    Download an image from image_url and upload it to glance using import,
+    validate and return a resource pointer.
+
+    :param glance: Authenticated glanceclient
+    :type glance: glanceclient.Client
+    :param image_url: URL to download image from
+    :type image_url: str
+    :param image_name: display name for new image
+    :type image_name: str
+    :param image_cache_dir: Directory to store image in before uploading. If it
+        is not passed, or is None, then a tmp directory is used.
+    :type image_cache_dir: Option[str, None]
+    :returns: glance image pointer
+    :rtype: glanceclient.common.utils.RequestIdProxy
+    """
+    if image_cache_dir is None:
+        image_cache_dir = tempfile.gettempdir()
+
+    logging.debug('Creating glance image '
+                  '({})...'.format(image_name))
+
+    img_name = os.path.basename(urllib.parse.urlparse(image_url).path)
+    local_path = os.path.join(image_cache_dir, img_name)
+
+    if not os.path.exists(local_path):
+        logging.info('Downloading {} ...'.format(image_url))
+        download_image(image_url, local_path)
+    else:
+        logging.info('Cached image found at {} - Skipping download'.format(
```

Comment:
you're duplicating code from `create_image()`. Can we refactor in order not to duplicate code?